pyogrio/tests/test_geopandas_io.py

```diff
@@ -1329,15 +1394,19 @@ def test_custom_crs_io(tmpdir, naturalearth_lowres_all_ext):
     assert df.crs.equals(expected.crs)
 
 
-def test_write_read_mixed_column_values(tmp_path):
+def test_write_read_mixed_column_values(request, tmp_path, use_arrow):
+    if use_arrow:
+        request.node.add_marker(
+            pytest.mark.xfail(reason="Arrow does not support mixed columns")
```

Comment:
Do we have reason to believe this behavior will change in the future?  If not, I think it would be better to split the Arrow test into a separate test case that specifically checks that an exception of appropriate type is raised.

Reply:
Indeed, I don't think we should add support for this, so will split off and explicitly test it raises an error instead of using an xfail